src/state/actions/window.js

```diff
@@ -63,6 +82,23 @@ export function toggleWindowSideBarPanel(windowId, panelType) {
   return { type: ActionTypes.TOGGLE_WINDOW_SIDE_BAR_PANEL, windowId, panelType };
 }
 
+/**
+ * popOutCompanionWindow - action creator
+ *
+ * @param  {String} windowId
+ * @param  {String} panelType The type of panel content to be rendered
+ *                            in the companion window (e.g. info, canvas_navigation)
+ * @param  {String} position The position of the companion window to
+ *                           set content for (e.g. right, bottom)
+ * @memberof ActionCreators
+ */
+export function popOutCompanionWindow(windowId, panelType, position) {
+  return ((dispatch) => {
+    dispatch(setWindowCompanionWindow(windowId, panelType, position));
+    dispatch(toggleWindowSideBarPanel(windowId, 'closed'));
```

Comment:
Not sure if this is the best way to go about this, but, it seems to work. ¯\\_(ツ)_/¯

Reply:
Another way to do it is to make this state change in the reducers. I can understand the urge to avoid repeating the state merging logic, so to DRY it up, I believe a utility function can be written and re-used in each of the appropriate reducers. The utility to merge the companion window into state and the utility to toggle the sidebar panel state would both be called in the reducer that receives popOutCompanionWindow. That way the action log would be clean (only the original intended action is fired), and the side effects would be removed.